### PR TITLE
Commitseal fix

### PR DESCRIPTION
**Pull request by yazzaoui**
Solves Committed Seals issue of https://github.com/clearmatics/autonity/issues/347

### CODE REVIEW

File: consensus/tendermint/core/core.go [hunk 1]
```diff
@@ -243,8 +243,10 @@ func (c *core) commit() {
 			committedSeals[i] = make([]byte, types.BFTExtraSeal)
 			copy(committedSeals[i][:], v.CommittedSeal[:])
 		}
-
-		if err := c.backend.Commit(*proposal.ProposalBlock, committedSeals); err != nil {
+		h := proposal.ProposalBlock.Header()
+		h.Round = c.currentRoundState.Round()
```
Comment (yazzaoui) [inline consensus/tendermint/core/core.go]: Would need a bit of refactoring, maybe moving that to `backend.Commit`

File: consensus/tendermint/core/core.go [hunk 2]
```diff
@@ -389,9 +391,10 @@ func (c *core) Quorum(i int) bool {
 }
 
 // PrepareCommittedSeal returns a committed seal for the given hash
-func PrepareCommittedSeal(hash common.Hash) []byte {
+func PrepareCommittedSeal(hash common.Hash, round *big.Int, height *big.Int) []byte {
 	var buf bytes.Buffer
+	buf.Write(round.Bytes())
+	buf.Write(height.Bytes())
```
Comment (yazzaoui) [inline consensus/tendermint/core/core.go]: Note that the height information is not strictly needed.

File: consensus/tendermint/core/precommit_test.go [hunk 3]
```diff
@@ -285,7 +285,7 @@ func TestHandlePrecommit(t *testing.T) {
 		}
 
 		backendMock := NewMockBackend(ctrl)
-		backendMock.EXPECT().Commit(*proposal.ProposalBlock, gomock.Any()).Return(nil)
+		backendMock.EXPECT().Commit(proposal.ProposalBlock, gomock.Any(), gomock.Any()).Return(nil)
```
Comment (JekaMas) [inline consensus/tendermint/core/precommit_test.go]: Are we waiting here 'any' or particular parameter?

Comment (yazzaoui) [reply consensus/tendermint/core/precommit_test.go]: yes the round and height number must match with the precommited seals, will update the test.

**Pull request merged**